share/spice/pokemon/data/pokemon_data.js

```diff
@@ -0,0 +1,118 @@
+(function (env) {
+    "use strict";
+
+    var ID = 'pokemon',
+        DESCRIPTION_ENDPOINT = '/js/spice/pokemon/description/{id}',
+        POKEAPI_SPRITE_URL = 'http://pokeapi.co/media/img/{id}.png',
+        INFOBOX_PROPS = ['hp', 'attack', 'defense', 'height', 'weight', 'speed'];
+
+    /**
+     *  [ Pokemon::Data ]
+     */
+    env.ddg_spice_pokemon_data = function(api_result){
+        if (!api_result) {
+            return Spice.failed('pokemon');
+        }
+
+        Spice.add({
+            id: ID,
+            name: 'Pokedex',
+            data: api_result,
+            meta: {
+                sourceName: 'pokeapi.co',
+                sourceUrl: 'http://pokeapi.co/',
+                sourceIconUrl: 'http://pokeapi.co/static/favicon.ico'
+            },
+            normalize: function(item) {
+                return {
+                    title: item.name,
+                    image: getSprite.call(item),
+                    infoboxData: getInfoboxData.call(item),
+                    eggGroups: getCollectionNames.call(item, 'egg_groups'),
+                    types: getCollectionNames.call(item, 'types')
+                };
+            },
+            templates: {
+                group: 'info',
+                options: {
+                    content: Spice.pokemon_data.content,
+                    moreAt: true
+                }
+            }
+        });
+
+        if( api_result.descriptions.length > 0 ) {
+            fetchDescription(api_result.descriptions);
+        }
+    };
+
+    /**
+     *  [ Pokemon::Description ]
+     */
+    env.ddg_spice_pokemon_description = function(api_result){
```

Comment:
@AkA84 great question. Originally we didn't have jQuery at our disposal so we relied on multiple callbacks. We'd use a function to create a script tag on the page which would cause the additional callback(s) to fire.

Now that we have jQuery, `$.getJSON` is simpler to use IMO and we now have the events which means a lot of the older IA just need to be updated and simplified.